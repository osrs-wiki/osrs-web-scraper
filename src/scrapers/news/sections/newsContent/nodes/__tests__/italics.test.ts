import { MediaWikiBuilder } from "@osrs-wiki/mediawiki-builder";
import parse from "node-html-parser";

import italicsParser from "../italics";

describe("italics node", () => {
  test("Italic text should parse and render", () => {
    const root = parse("<i>test</i>");
    const builder = new MediaWikiBuilder();
    builder.addContents([italicsParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Italic text should parse and render with a link", () => {
    const root = parse(
      "<i>testing this <a href='https://example.com'>link</a></i>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([italicsParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });

  test("Italic text should parse and render with a link and bold", () => {
    const root = parse(
      "<i>testing this <a href='https://example.com'>link</a> and <b>bold</b></i>"
    );
    const builder = new MediaWikiBuilder();
    builder.addContents([italicsParser(root.firstChild)].flat());
    expect(builder.build()).toMatchSnapshot();
  });
});
