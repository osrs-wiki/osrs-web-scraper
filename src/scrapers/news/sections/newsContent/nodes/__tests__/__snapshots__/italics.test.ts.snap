// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`italics node Italic text should parse and render 1`] = `"''test''"`;
